### PR TITLE
Feature/ab#89 legal guardian email change validation

### PR DESCRIPTION
**Pull request by vyruz1986**

### CODE REVIEW

File: Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts [hunk 1]
```diff
@@ -9,10 +9,9 @@ import {
    MatPaginator,
    MatSort,
    MatDialog,
-   MAT_DATE_FORMATS,
+   MatSnackBar,
 } from '@angular/material';
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts:13]: Ooh, painful. We will need this for our date formatting

File: Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts [hunk 2]
```diff
@@ -68,10 +68,22 @@ export class OverviewComponent implements OnInit, AfterViewInit {
 
       dialogRef.componentInstance.submitEvent.subscribe((result: CareUser) => {
          // Update the Careuser
-         this.careUserService.updateUser(result).subscribe(res => {
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts:71]: Remember our discussion regarding javascript and by reference passing? It seems that javascript does this for objects and arrays, and only (obviously) for primitive types by value. So yes, updating the row values (even in a popup), means actually updating the row.

File: Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts [hunk 3]
```diff
@@ -123,4 +145,24 @@ export class OverviewComponent implements OnInit, AfterViewInit {
...
+   handleApiError(err: any) {
+      if (typeof err === 'string') {
+         this.snackBar.open(`⚠ ${err}`, 'OK');
+      } else if (typeof err === 'object' && err !== null) {
+         let messages = [];
+         for (var k in err) {
+            messages.push(err[k]);
+         }
+         this.snackBar.open(
+            `⚠ Er zijn fouten opgetreden bij het opslagen:\n${messages.join(
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts]: Opslagen should be "opslaan"

File: Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singerevent-overview/singerevent-overview.component.ts [hunk 4]
```diff
@@ -152,4 +176,24 @@ export class SingerEventOverviewComponent implements OnInit, AfterViewInit {
          )
          .subscribe();
    }
+
+   handleApiError(err: any) {
```
Comment (BerendWouters) [inline Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singerevent-overview/singerevent-overview.component.ts:180]: Fancy!

**Review by BerendWouters — CHANGES_REQUESTED** [review]

File: Singer.API/ClientApp/package.json [hunk 5]
```diff
@@ -52,11 +52,11 @@
       "karma-coverage-istanbul-reporter": "~1.4.2",
       "karma-jasmine": "~1.1.1",
       "karma-jasmine-html-reporter": "^0.2.2",
-      "node-sass": "^4.12.0",
       "standard-version": "^7.0.0",
       "typescript": "~3.2.4"
    },
    "optionalDependencies": {
+      "node-sass": "^4.9.3",
```
Comment (BerendWouters) [inline Singer.API/ClientApp/package.json]: What happened here? It causes the build to break.

**Review by BerendWouters — CHANGES_REQUESTED** [review]

File: Singer.API/ClientApp/package.json (hunk 5 above)
Comment (vyruz1986) [reply Singer.API/ClientApp/package.json]: Thanks for fixing :)

Comment (vyruz1986) [reply Singer.API/ClientApp/package.json]: Thanks for fixing :)

File: Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts (hunk 3 above)
Comment (vyruz1986) [reply Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts]: Fixed

File: Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts (hunk 2 above)
Comment (vyruz1986) [reply Singer.API/ClientApp/src/app/modules/admin/components/careusers/overview/overview.component.ts:71]: It has been fixed

**Review by BerendWouters — APPROVED** [review]

**Pull request merged**